front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/SnapshotsController.groovy

```diff
@@ -57,6 +57,20 @@ class SnapshotsController {
         return snapshotDAO.findById(id)
     }
 
+    @PostAuthorize("hasPermission(returnObject.application, 'APPLICATION', 'READ')")
+    @RequestMapping(value = "/{id:.+}/{timestamp:.+}", method = RequestMethod.GET)
+    Snapshot getVersionByTimestamp(@PathVariable String id,
+                        @PathVariable String timestamp,
+                        @RequestParam(value = "limit", defaultValue = "20") int limit) {
+        def creationTime = timestamp.toLong()
+        for (Snapshot snapshot: snapshotDAO.getHistory(id, limit)) {
```

Comment:
This should probably be
` snapshotDAO.getHistory(id, limit)?.find { snapshot -> snapshot.timestamp == creationTime }`